src/mach/templates/partials/endpoints/gcp_api_gateway.tf

```diff
@@ -0,0 +1,7 @@
+
+
+resource "google_api_gateway_api" "{{ endpoint.key|slugify }}_gateway" {
```

Comment:
I think we should have a look at the `api_gateway_gateway` resource, as it seems to be the missing link for providing a central API gateway that sits in front of multiple APIs?

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/api_gateway_gateway